### DISCUSSION TITLE
API keys for the free Open-Meteo API

### DISCUSSION THREAD

**Discussion by rany2** [open-meteo/open-meteo#843] [General]
I'm not sure if it's a feasible idea, but some users that use [my HA integration for PV forecast](https://github.com/rany2/ha-open-meteo-solar-forecast) face rate-limits because they are behind IPv4 CGNAT and have neighbors that also use Open-Meteo. Maybe it could be solved by allowing users to register for accounts and have API keys.

**Comment by patrick-zippenfenig** [comment]
Continued here: https://github.com/open-meteo/open-meteo/discussions/853#discussioncomment-9790613
